rllib/evaluation/collectors/agent_collector.py

```diff
@@ -204,8 +243,11 @@ def add_action_reward_next_obs(self, input_values: Dict[str, TensorType]) -> Non
         self.buffers[SampleBatch.UNROLL_ID][0].append(self.unroll_id)
 
         for k, v in values.items():
+            self._check_view_requirement(k, v)
```

Comment:
maybe hide this behind a ``AlgorithmConfig.debugging()`` flag?
I have some performance concerns about enabling this by default all the time.
also, I think a lot of times, k will not have a corresponding VR in the map?
this may be the reason all CI tests are broken right now.

Reply:
It's indeed why lots of tests **where** broken and I did not want to include it. @kouroshHakha wanted this check here but I had to make it a warning ultimately because we sometimes don't even have spaces for VRs atm.

_check_view_requirement checks depending on the following clause now:
`if (
            log_once(
                f"view_requirement_"
                f"{view_requirement_name}_checked_in_agent_collector"
            )
            and view_requirement_name in self.view_requirements
        ):`

Reply:
So this is a warning and it only warns once.

Reply:
ok